tools/stop_machine_aws.py

```diff
@@ -1,31 +1,32 @@
-import subprocess
 import boto3 
-import os
-import sys
-import logging
-import argparse
 import datetime
+import re
 
 def stop(instance, logger):
     if instance.state['Name'] != 'stopped':
             instance.stop()
             logger.info("Stop the instance - Id: {0}, Type: {1}, Public IPv4: {2}, State: {3}".format(instance.id, instance.instance_type, instance.public_ip_address, instance.state))
 
+def is_match_name_instance(names, instance):
+    for name in names:
+        if re.search(name, instance.tags[0]['Value']):
+            return True
+    return False
+
 def stop_machines(type_machine, name_machines, logger):
     logger.info('Stop machines...')
     try:
         ec2 = boto3.resource('ec2', region_name='us-west-2')
-    except:
-        logger.info(f'{datetime.datetime.now()}: Can\'t connect to ec2 with boto3 for stop the machines')
-        return
+    except botocore.exceptions.ClientError as error:
+        logger.error(f'{datetime.datetime.now()}: Error {error}  - Can\'t connect to ec2 with boto3 for stop the machines')
 
     list_instances = ec2.instances.all()
 
-    types =type_machine.split('_') if type_machine else '' 
-    names =name_machines.split('_') if name_machines else ''
-    
+    types = type_machine.split(',') if type_machine else '' 
+    names = name_machines.split(',') if name_machines else ''
+
     for instance in list_instances:
-        if not instance.state['Name'] == 'stopped' and (types[0]=='all' or instance.instance_type in types) and (names[0]=='all' or instance.tags[0]['Value'] in names):
+        if not instance.state['Name'] == 'stopped' and (((types and names) and (instance.instance_type in types or is_match_name_instance(names, instance))) or ((not types or instance.instance_type in types) and (not names or is_match_name_instance(names, instance)))):
```

Comment:
Refactor long one line conditions to a multi-line function.
Always prefer clearer and simpler code (unless there are specific performance requirements).

```python
def is_stop_instance(instance):
  if instance.state['Name'] == 'stopped':
    return False;

  if types and instance.instance_type not in types:
    return False;

  if names and not is_match_name(names, instance):
    return False;

  return True;
```